Account/FirefoxAccount.swift

```diff
@@ -21,7 +26,7 @@ let AccountSchemaVersion = 2
 ///
 /// Non-sensitive but persistent data should be maintained outside of
 /// the account itself.
-open class FirefoxAccount {
+open class FirefoxAccount: CollectionChangedNotifier {
```

Comment:
Nope! The delegate is an artifact of the moment of beginning a sync.

This is a good place for the actual notification logic to live, but that doesn't mean `FirefoxAccount` is a `SyncCollectionChangeNotificationDelegate` (which is what this would be called in Java).

There's a triangular relationship between an account, a `SyncManager`, and a synchronizer.